### ISSUE TITLE
Test which passes on mock fails on real (base64)

### ISSUE THREAD

**Issue by adamtheturtle** [adamtheturtle/vws-python-mock#461]
In `test_add_target.py` (likely similar in `test_update_target.py`:

#### In `TestImage`

```python
    def test_not_base64_encoded(
        self,
        vuforia_database: VuforiaDatabase,
    ) -> None:
        """
        If the given image is not decodable as base64 data then a `Fail`
        result is returned.
        """
        not_base64_encoded = 'a'

        with pytest.raises(binascii.Error):
            base64.b64decode(not_base64_encoded)

        data = {
            'name': 'example_name',
            'width': 1,
            'image': not_base64_encoded,
        }

        response = add_target_to_vws(
            vuforia_database=vuforia_database,
            data=data,
        )

        assert_vws_failure(
            response=response,
            status_code=codes.UNPROCESSABLE_ENTITY,
            result_code=ResultCodes.FAIL,
        )
```

#### In `TestApplicationMetadata`

```python
    def test_not_base64_encoded(
        self,
        vuforia_database: VuforiaDatabase,
        image_file_failed_state: io.BytesIO,
    ) -> None:
        """
        A string which is not base64 encoded is not valid application metadata.
        """
        image_data = image_file_failed_state.read()
        image_data_encoded = base64.b64encode(image_data).decode('ascii')

        not_base64_encoded = 'a'

        with pytest.raises(binascii.Error):
            base64.b64decode(not_base64_encoded)

        data = {
            'name': 'example_name',
            'width': 1,
            'image': image_data_encoded,
            'application_metadata': not_base64_encoded,
        }

        response = add_target_to_vws(
            vuforia_database=vuforia_database,
            data=data,
        )

        assert_vws_failure(
            response=response,
            status_code=codes.UNPROCESSABLE_ENTITY,
            result_code=ResultCodes.FAIL,
        )
```

**Comment by adamtheturtle** [comment]
Base64 encoded strings are of length of a multiple of 4 - there is a different error when the given string is of length multiple 4 and not.

**Comment by adamtheturtle** [comment]
Seems like the above is wrong - it seems like for `aa` and `a` - our checks say that these are not base64 encoded strings, but Vuforia treats them as if they are. Perhaps there is automatic padding?

**Comment by adamtheturtle** [comment]
I have learned the following.

When using `base64.b64decode(s, validate=True)`, there are multiple errors in the error messages for the `binascii.Error` exception.

These include (at least):

* The wrong type of character used, e.g. `s = '"'`, "binascii.Error: Non-base64 digit found" - Vuforia does not accept this type of string
* The length of `s` `% 4 == 1`: binascii.Error: Invalid base64-encoded string: number of data characters (1) cannot be 1 more than a multiple of 4 - Vuforia handles this by ignoring the last digit
* The length of `s` `% 4` is not 1 or 0: Vuforia pads this with 0s

I have also learned that sometimes, decoding and encoding a string to base64 does not lead to the original string. For example:

```
print(base64.b64encode(base64.b64decode('XN==', validate=True)))
```

Prints `XA==` and Vuforia's query returns the latter in the application metadata if the former is given.
